AndroidSDKCore/src/main/java/com/leanplum/internal/RequestOldUtil.java

```diff
@@ -0,0 +1,30 @@
+package com.leanplum.internal;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import com.leanplum.Leanplum;
+import com.leanplum.utils.SharedPreferencesUtil;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static com.leanplum.internal.RequestOld.LEANPLUM;
+import static com.leanplum.internal.RequestOld.UUID_KEY;
+
+public class RequestOldUtil {
+
+    public void setNewBatchUUID(List<Map<String, Object>> requests) {
+        String uuid = UUID.randomUUID().toString();
+        Context context = Leanplum.getContext();
+        SharedPreferences preferences = context.getSharedPreferences(
+                LEANPLUM, Context.MODE_PRIVATE);
+        SharedPreferences.Editor editor = preferences.edit();
+        editor.putString(Constants.Defaults.UUID_KEY, uuid);
```

Comment:
SharedPrefs for storing BatchId, but it it used for some reason? Do we have a get batchId from sharedprefs? Wanted to understand the use case for sharedprefs.
Also, if we are doing getSharedPrefs for some reason, not sure when. Will it be a problem, is say the API fall fails?

Reply:
The batchId is gotten from SharedPrefs when new requests are added. https://github.com/Leanplum/Leanplum-Android-SDK/blob/b86e546598f6f28c8f7809567327204484c3dea8/AndroidSDKCore/src/main/java/com/leanplum/internal/RequestOld.java#L158

Reply:
I have not changed anything about API call failures in this PR. I am not sure what the exact mechanism is but you can check it out in `RequestOld.java` 
I know that when we send a batch, is when we change the UUID that's stored in sharedprefs. So that's the model I am following